primer/src/Primer/Core/DSL.hs

```diff
@@ -159,6 +160,10 @@ letType v t e = LetType <$> meta <*> pure v <*> t <*> e
 case_ :: MonadFresh ID m => m Expr -> [m CaseBranch] -> m Expr
 case_ e brs = Case <$> meta <*> e <*> sequence brs <*> pure CaseExhaustive
 
+-- | A non-exhaustive case
+caseFB_ :: MonadFresh ID m => m Expr -> [m CaseBranch] -> m Expr -> m Expr
+caseFB_ e brs fb = Case <$> meta <*> e <*> sequence brs <*> (CaseFallback <$> fb)
```

Comment:
We leave `case_` alone to avoid churn. (I should mention this in the commit message)